data_pipeline/civicrm_email_report/civicrm_email_report.py

```diff
@@ -0,0 +1,65 @@
+import json
+import datetime
+import logging
+from typing import Iterable
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_current_timestamp():
+    return datetime.datetime.utcnow().isoformat()
+
+
+def get_mailing_id_dict(mail_id: int) -> str(dict):
+    return json.dumps({'mailing_id': mail_id})
+
+
+def get_connection_parameters(mail_id: int, api_key, site_key):
+    return {
+        "entity": "Mailing",
+        "action": "stats",
+        "json": get_mailing_id_dict(mail_id),
+        "api_key": api_key,
+        "key": site_key
+    }
+
+
+def get_email_report(
+    url: str,
+    mail_id: int,
+    api_key: str,
+    site_key: str
+) -> dict:
+    response = requests.post(url=url, data=get_connection_parameters(mail_id, api_key, site_key))
+    response.raise_for_status()
+    dict_response = json.loads(response.text)
```

Comment:
yes, using the `json` library is much better. I believe the response should also have a `json` function?

Reply:
Yes normally I am using it too bu unfortunately it did not work for this case. json function returning something totally different like` <bound method Response.json of <Response [200]>>`

Reply:
This `<bound method...` would suggest to me that it was the reference to the `json` function rather than the return value.
Maybe you didn't call it?

Reply:
Probably I misunderstand, what did you mean by not calling? I was thinking the reason is about using `post` request instead of `get` but if I need to call something else I am not aware of it?